intake/__init__.py

```diff
@@ -126,6 +126,8 @@ def open_catalog(uri=None, **kwargs):
     intake.open_intake_remote
     """
     driver = kwargs.pop('driver', None)
+    if hasattr(uri, "__fspath__"):
+        uri = uri.__fspath__()
```

Comment:
I think that `os.path(uri)` would be better here. It handles the dispatch. And this reads like `a.__len__()` vs `len(a)`.

Reply:
I specifically test for `__fspath__`, though. What is the best way to test for something `Path`-like? Seems best to avoid `isinstance` for, e.g., the possible future fsspec-pathlike compatibility layer.

Reply:
(actually, the simplest conversion from Path to str is `str()`)